app/controllers/recipes_controller.rb

```diff
@@ -137,6 +137,15 @@ def unbookmark  # saved
     end
   end
 
+  def tagged
+    if params[:tag].present?
+      @recipes = Recipe.tagged_with(params[:tag])
+      @tag = params[:tag]
+    else
+      @recipes = Recipe.all
+    end
+  end
+
```

Comment:
tagged method should be allowed to be called when user is not signed in.
before_action :authenticate_user!, except: [:index, :show, :tagged]